test/AutoDiff/validation-test/storeborrow.swift

```diff
@@ -0,0 +1,156 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import DifferentiationUnittest
+
+var StoreBorrowAdjTest = TestSuite("StoreBorrowAdjTest")
+
+public struct ConstantTimeAccessor<Element>: Differentiable where Element: Differentiable, Element: AdditiveArithmetic {
+    /// A variably-sized representation of the tangent vector for the array, only populating with as many zeros or
+    /// values as needed. This allows for empty arrays in the .zero case, or small ones if only the first few values are
+    /// active. A fully-sparse Dictionary-based tangent vector suffers from extremely slow additions, which are the
+    /// most common operations for tangent vectors. This provides a balance between sparsity and quick addition.
```

Comment:
We can probably remove the comments in this test case that are left over from its original implementation, in order to simplify the test.

Reply:
Or maybe leave as-is to show the details of implementation? Anyway, I'd let you to decide whether to keep them or remove

Reply:
+1 to removing semantic info from tests that's not needed for reproducing errors.

You could create and reference an issue with more context (e.g. `ConstantTimeAccessor`) as has been done in some AutoDiff testcase names, or you could leave a code comment with a link to the context.